script.js

```diff
@@ -1,4 +1,312 @@
+        cardName = 'king';
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+// Get sum of cards in hand
+var getHandSum = function (hand) {
+  console.log(`hand in getHandSum is `)
+  console.log(hand)
+  var numAcesInHand = 0;
+  var sumLimit= 21;
+  var sum = 0;
+  for (let i = 0; i < hand.length; i += 1) {
+    var currCard = hand[i];
+    // If card rank is 2-10, value is same as rank
+    if (currCard.rank >= 2 && currCard.rank <= 10) {
+      sum += currCard.rank;
+      // If card rank is 11-13, i.e. Jack, Queen, or King, value is 10
+    } else if (currCard.rank >= 11 && currCard.rank <= 13) {
+      sum += 10;
+      // If card is Ace, value is 11 by default
+    } else if (currCard.rank === 1) {
+      numAcesInHand += 1;
+      sum += 11;
+    }
+  }
+  // If sum is greater than sum limit and hand contains Aces, convert Aces from value of 11
+  // to value of 1, until sum is less than or equal to sum limit or there are no more Aces.
+  if (sum > sumLimit && numAcesInHand > 0) {
+    for (let i = 0; i < numAcesInHand; i += 1) {
+      sum -= 10;
+      // If the sum is less than sumLimit before converting all Ace values from
+      // 11 to 1, break out of the loop and return the current sum.
+      if (sum <= sumLimit) {
+        break;
+      }
+    }
+  }
+  console.log(sum)
+  return sum;
+};
+// function to convert suit word to emoji
+var convertSuitWordToEmoji = function (suitWord) {
+  if (suitWord == 'spades') {
+    return '♠️';
+  }
+  if (suitWord == 'hearts') {
+    return '♥️';
+  }
+  if (suitWord == 'clubs') {
+    return '♣️';
+  }
+  if (suitWord == 'diamonds') {
+    return '♦️';
+  }
```

Comment:
Nice, showed you understand the code well enough to manipulate the symbol and improve the game UI